util/printchplenv

```diff
@@ -39,260 +109,278 @@ def print_mode(mode='list', anonymize=False):
-        link_args_3p.extend(chpl_3p_gmp_configs.get_link_args(gmp))
-
-        print_var('  CHPL_HWLOC_UNIQ_CFG_PATH',
-                  chpl_3p_hwloc_configs.get_uniq_cfg_path(),
-                  mode, filters=('runtime',))
-        link_args_3p.extend(chpl_3p_hwloc_configs.get_link_args(hwloc))
-
-        print_var('  CHPL_JEMALLOC_UNIQ_CFG_PATH',
-                  chpl_3p_jemalloc_configs.get_uniq_cfg_path(),
-                  mode, filters=('runtime',))
-        link_args_3p.extend(chpl_3p_jemalloc_configs.get_link_args(jemalloc))
-
-        print_var('  CHPL_LIBUNWIND_UNIQ_CFG_PATH',
-                  chpl_3p_libunwind_configs.get_uniq_cfg_path(),
-                  mode, filters=('runtime',))
-        link_args_3p.extend(chpl_3p_libunwind_configs.get_link_args(unwind))
-
-        print_var('  CHPL_MASSIVETHREADS_UNIQ_CFG_PATH',
-                  chpl_3p_massivethreads_configs.get_uniq_cfg_path(),
-                  mode, filters=('runtime',))
-        if tasks == 'massivethreads':
+        ENV_VALS['  CHPL_MASSIVETHREADS_UNIQ_CFG_PATH'] = chpl_3p_massivethreads_configs.get_uniq_cfg_path()
+        if chpl_tasks.get() == 'massivethreads':
             link_args_3p.extend(chpl_3p_massivethreads_configs.get_link_args())
 
-        print_var('  CHPL_QTHREAD_UNIQ_CFG_PATH',
-                  chpl_3p_qthreads_configs.get_uniq_cfg_path(),
-                  mode, filters=('runtime',))
-        if tasks == 'qthreads':
+        ENV_VALS['  CHPL_QTHREAD_UNIQ_CFG_PATH'] = chpl_3p_qthreads_configs.get_uniq_cfg_path()
+        if chpl_tasks.get() == 'qthreads':
             link_args_3p.extend(chpl_3p_qthreads_configs.get_link_args())
 
-        print_var('  CHPL_RE2_UNIQ_CFG_PATH',
-                  chpl_3p_re2_configs.get_uniq_cfg_path(),
-                  mode, filters=('runtime',))
-        if regexp == 're2':
+        ENV_VALS['  CHPL_RE2_UNIQ_CFG_PATH'] = chpl_3p_re2_configs.get_uniq_cfg_path()
+        if chpl_regexp.get() == 're2':
             link_args_3p.extend(chpl_3p_re2_configs.get_link_args())
 
-        link_args_3p_dedup=[]
-        for arg in reversed(link_args_3p):
-            if arg not in link_args_3p_dedup:
-                link_args_3p_dedup.append(arg)
-        print_var('  CHPL_THIRD_PARTY_LINK_ARGS',
-                  ' '.join(reversed(link_args_3p_dedup)),
-                  mode, filters=('runtime',))
+        # Remove duplicates, keeping last occurrence and preserving order
+        # e.g. "-lhwloc -lqthread -lhwloc ..." -> "-lqthread -lhwloc ..."
+        seen = set()
+        link_args_3p_dedup = [arg for arg in reversed(link_args_3p)
+                              if not (arg in seen or seen.add(arg))]
+        ENV_VALS['  CHPL_THIRD_PARTY_LINK_ARGS'] = ' '.join(reversed(link_args_3p_dedup))
+
+
+"""Return non-empty string if var is set via environment or chplconfig"""
+def user_set(env):
+    env_set = overrides.get_environ(env, '')
+    config_set = overrides.get_chplconfig(env.strip(), '')
```

Comment:
It seems odd that you needed env.strip only for the chplconfig version? I'd probably put env.strip at the start of this routine and use it for both.